pallets/nfts/src/tests.rs

```diff
@@ -1942,53 +2330,307 @@ fn approving_multiple_accounts_works() {
+			Nfts::transfer(RuntimeOrigin::signed(account(5)), 0, 42, account(8)),
+			Error::<Test>::NoPermission
+		);
+	});
+}
+
+#[test]
+fn approvals_limit_works() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			account(1),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(account(1)),
+			0,
+			42,
+			account(2),
+			default_item_config()
+		));
+
+		for i in 3..13 {
+			assert_ok!(Nfts::approve_transfer(
+				RuntimeOrigin::signed(account(2)),
+				0,
+				42,
+				account(i),
+				None
+			));
+		}
+		// the limit is 10
+		assert_noop!(
+			Nfts::approve_transfer(RuntimeOrigin::signed(account(2)), 0, 42, account(14), None),
+			Error::<Test>::ReachedApprovalLimit
+		);
+	});
+}
+
+#[test]
+fn approve_collection_transfer_works() {
+	new_test_ext().execute_with(|| {
+		let (collection, locked_collection) = (0, 1);
+		let item = 42;
+		let collection_owner = account(1);
+		let item_owner = account(2);
+		let delegate = account(3);
+
+		Balances::make_free_balance_be(&item_owner, 100);
+		Balances::make_free_balance_be(&delegate, 100);
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			collection_owner.clone(),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			collection_owner.clone(),
+			default_collection_config()
+		));
+
+		// Approve collection without items, throws error `Error::NoItemOwned`.
+		assert_noop!(
+			Nfts::approve_collection_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				collection,
+				delegate.clone(),
+				None
+			),
+			Error::<Test>::NoItemOwned
+		);
+
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(collection_owner.clone()),
+			collection,
+			item,
+			item_owner.clone(),
+			default_item_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(collection_owner.clone()),
+			locked_collection,
+			item,
+			item_owner.clone(),
+			default_item_config()
+		));
+
+		// Throws error `Error::ItemsNonTransferable`.
+		assert_ok!(Nfts::lock_collection(
+			RuntimeOrigin::signed(collection_owner),
+			locked_collection,
+			CollectionSettings::from_disabled(CollectionSetting::TransferableItems.into())
+		));
+		assert_noop!(
+			Nfts::approve_collection_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				locked_collection,
+				delegate.clone(),
+				None
+			),
+			Error::<Test>::ItemsNonTransferable
 		);
 
-		assert_ok!(Nfts::transfer(RuntimeOrigin::signed(account(4)), 0, 42, account(6)));
+		// Approve unknown collection, throws error `Error::NoItemOwned`.
 		assert_noop!(
-			Nfts::transfer(RuntimeOrigin::signed(account(3)), 0, 42, account(7)),
-			Error::<Test>::NoPermission
+			Nfts::approve_collection_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				2,
+				delegate.clone(),
+				None
+			),
+			Error::<Test>::NoItemOwned
 		);
+
+		// Approval expires after `deadline`.
```

Comment:
This shouldn't be in the test for `approve_collection_transfer`. Unit tests should be written specific to its function. Checking the `check_approval` is not doing that. Please refactor this test to only testing the items, errors etc specific to this function.

Reply:
Removed in the next commit.